diesel/src/deserialize.rs

```diff
@@ -420,12 +420,10 @@ pub trait FromSql<A, DB: Backend>: Sized {
     ///
     /// If your custom type supports null values you need to provide a
     /// custom implementation.
-    #[inline(always)]
     fn from_nullable_sql(bytes: Option<backend::RawValue<'_, DB>>) -> Result<Self> {
-        match bytes {
-            Some(bytes) => Self::from_sql(bytes),
-            None => Err(Box::new(crate::result::UnexpectedNullError)),
-        }
+        bytes
```

Comment:
I think we could use [`map_or_else`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.map_or_else) if we want a shorter writing, but otherwise I find the previous writing easier to read